cns/restserver/internalapi.go

```diff
@@ -96,22 +96,22 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 
 	// check if the version is valid and save it to service state
 	for ncid, nc := range ncsToBeAdded {
-		var (
-			versionURL = fmt.Sprintf(nmagent.GetNetworkContainerVersionURLFmt,
-				nmagent.WireserverIP,
-				nc.PrimaryInterfaceIdentifier,
-				nc.NetworkContainerid,
-				nc.AuthorizationToken)
-			w = httptest.NewRecorder()
-		)
-
-		ncVersionURLs.Store(nc.NetworkContainerid, versionURL)
+		nmaReq := nmagent.NCVersionRequest{
+			AuthToken:          nc.AuthorizationToken,
+			NetworkContainerID: nc.NetworkContainerid,
+			PrimaryAddress:     nc.PrimaryInterfaceIdentifier,
+		}
+
+		ncVersionURLs.Store(nc.NetworkContainerid, nmaReq)
 		waitingForUpdate, _, _ := service.isNCWaitingForUpdate(nc.Version, nc.NetworkContainerid)
 
 		body, _ = json.Marshal(nc)
 		req, _ = http.NewRequest(http.MethodPost, "", bytes.NewBuffer(body))
 		req.Header.Set(common.ContentType, common.JsonContent)
+
+		w := httptest.NewRecorder()
```

Comment:
I know this is just moving existing lines, but this is... interesting, to say the least. i suppose its a symptom of all routing + business logic living in the same function.

Reply:
Not a fan of `httptest` in something that's not a test anyway... but that's a battle for another day.